rest/server_context_test.go

```diff
@@ -163,3 +165,197 @@ func (client *MockClient) RespondToGET(url string, response *http.Response) {
 	tripper := client.Transport.(*mockTripper)
 	tripper.getURLs[url] = response
 }
+
+func TestRemoveDatabase(t *testing.T) {
+	server := "walrus:"
+	bucketName := "imbucket"
+	databaseName := "imdb"
+
+	serverConfig := &ServerConfig{CORS: &CORSConfig{}, AdminInterface: &DefaultAdminInterface}
+	serverContext := NewServerContext(serverConfig)
+	bucketConfig := BucketConfig{Server: &server, Bucket: &bucketName}
+	dbConfig := &DbConfig{BucketConfig: bucketConfig, Name: databaseName, AllowEmptyPassword: true}
+
+	dbContext, err := serverContext.AddDatabaseFromConfig(dbConfig)
+	assert.NoError(t, err, "No error while adding database to server context")
+	assert.Equal(t, server, dbContext.BucketSpec.Server)
+	assert.Equal(t, bucketName, dbContext.BucketSpec.BucketName)
+
+	status := serverContext.RemoveDatabase(databaseName)
+	assert.True(t, status, "Database should be removed from server context")
+	dbContext, err = serverContext.GetDatabase(databaseName)
+	assert.Nil(t, dbContext, "Database context should not be available!")
+}
+
+func TestAllDatabaseNames(t *testing.T) {
+	server := "walrus:"
+	bucketName := "imbucket"
+	var databaseNames []string
+
+	serverConfig := &ServerConfig{CORS: &CORSConfig{}, AdminInterface: &DefaultAdminInterface}
+	serverContext := NewServerContext(serverConfig)
+	bucketConfig := BucketConfig{Server: &server, Bucket: &bucketName}
+
+	for i := 0; i < 5; i++ {
+		databaseName := fmt.Sprintf("imdb%d", i)
+		databaseNames = append(databaseNames, databaseName)
+		dbConfig := &DbConfig{BucketConfig: bucketConfig, Name: databaseName, AllowEmptyPassword: true}
+		dbContext, err := serverContext.AddDatabaseFromConfig(dbConfig)
+
+		assert.NoError(t, err, "No error while adding database to server context")
+		assert.Equal(t, server, dbContext.BucketSpec.Server)
+		assert.Equal(t, bucketName, dbContext.BucketSpec.BucketName)
+		assert.ElementsMatch(t, databaseNames, serverContext.AllDatabaseNames())
+	}
+
+	databaseNames = removeElementWithValue(databaseNames, "imdb2")
+	databaseNames = removeElementWithValue(databaseNames, "imdb3")
+
+	status := serverContext.RemoveDatabase("imdb2")
+	assert.True(t, status, "Database should be removed from server context")
+	status = serverContext.RemoveDatabase("imdb3")
+	assert.True(t, status, "Database should be removed from server context")
+
+	assert.ElementsMatch(t, databaseNames, serverContext.AllDatabaseNames())
+}
+
+func removeElementWithIndex(s []string, index int) []string {
+	return append(s[:index], s[index+1:]...)
+}
+
+func removeElementWithValue(s []string, e string) []string {
+	return removeElementWithIndex(s, indexOf(s, e))
+}
+
+func indexOf(elements []string, e string) int {
+	for k, v := range elements {
+		if e == v {
+			return k
+		}
+	}
+	return -1 // Not found.
+}
+
+func captureOutput(f func()) string {
+	var buf bytes.Buffer
+	log.SetOutput(&buf)
+	defer func() { log.SetOutput(os.Stderr) }()
+	f()
+	return buf.String()
+}
```

Comment:
This is a red flag for anything that doesn't specifically involve logging testing :)
There will be a better way of asserting certain behaviour than searching the logs.

Reply:
Sure, let's remove this and rewrite in another way.